py/packages/corpora_commander/api/draft.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from typing import Any, Dict, List
+from uuid import UUID
+
+from corpora_ai.llm_interface import ChatCompletionTextMessage
+from django.shortcuts import get_object_or_404
+from pydantic import BaseModel
+
+from corpora_commander.models import Project
+
+from .llm_utils import build_llm
+from .router import router
+
+logger = logging.getLogger(__name__)
+
+
+class DraftSubsection(BaseModel):
+    subsection_id: UUID
+    content: str
+
+
+class DraftSection(BaseModel):
+    section_id: UUID
+    introduction: str
+    subsections: List[DraftSubsection]
+
+
+class DraftBookResponse(BaseModel):
+    sections: List[DraftSection]
+
+
+class DraftBookRequest(BaseModel):
+    provider: str
+    config: Dict[str, Any]
+    prompt: str = "Please draft the full book based on the outline and instructions provided."
+
+
+@router.post("/projects/{project_id}/draft", response=DraftBookResponse)
+def draft_book(request, project_id: UUID, payload: DraftBookRequest):
```

Comment:
Takes a long time, would be nice to have other modes/options?

- parallelize independent sections?
- parallel subsections?
- send to celery and SSE (server-side event) back the progress?